File: 11st/pro_10.java
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        Stack<Integer> s = new Stack<>();
        s.push(0);
        for(int i=1; i<prices.length; i++){
            while(!s.empty()&&prices[s.peek()]>prices[i]){
                answer[s.peek()]=i-s.peek();
                s.pop();
            }
            s.push(i);
        }
        while(!s.empty()){
            answer[s.peek()]=prices.length-s.peek()-1;
            s.pop();
        }
        return answer;
    }
}

File: 12nd/pro_10.cpp
#include <cmath>

using namespace std;

int solution(int n) {
    int answer = 1;
    for(int i=3; i<=n; i++){
        bool tmp=true;
        for(int j=2; j<=sqrt(i); j++){
            if(i%j==0){
                tmp=false; break;
            }
        }
        if(tmp){
            answer++;
        }
    }
    return answer;
}